packages/vue_frontend_rpc/test/backend-mock.js

```diff
@@ -0,0 +1,30 @@
+const ASYNC_NOOP = async () => {};
+
+const DefaultRPCFunctions = {
+  onFrontendReady: ASYNC_NOOP,
+  save: ASYNC_NOOP,
+};
+
+/**
+ * This function gets some function implementations that should be available when calling via RPC.
+ * @param userProvidedRpcApis - functions implementations
+ * @returns invoke mock function. When it will be called with the injected function implementation, this given mock implementation
+ * will be executed. Other non given functions will have default empty implementation.
+ */
+export function backendMock(userProvidedRpcApis) {
+  return async function invokeMock(funcName, params) {
+    const actualOpts = {};
+    for (const [rpcFuncName, rpcFuncDefault] of Object.entries(
+      DefaultRPCFunctions
+    )) {
+      // choose between user (optional) provided and default (NOOP)
+      actualOpts[rpcFuncName] =
+        userProvidedRpcApis[rpcFuncName] || rpcFuncDefault;
```

Comment:
we always know what the `rpcFuncDefault` value is, so why do we need `Object.entries`?

Reply:
We need it more for fetching the keys dynamically...but used it also for the values (rpcFuncDefault)